test/testother.cpp

```diff
@@ -3997,6 +3997,19 @@ class TestOther : public TestFixture {
         check("void f(bool a) { if( !(!a) != !(a) ) {}}");
         ASSERT_EQUALS("[test.cpp:1] -> [test.cpp:1]: (style) Opposite expression on both sides of '!='.\n", errout.str());
 
+
+        check("void foo(int i, const int n) { if ( i < n && i == n ) {} }");
+        ASSERT_EQUALS("[test.cpp:1] -> [test.cpp:1]: (style) Opposite expression on both sides of '&&'.\n", errout.str());
```

Comment:
hmm.. this message is not ideal. the expressions are not "opposite" as I define it.

However how do we write a message that is both readable and correct, I don't know. I think it would be more technically correct to say that it's a "logical disjuntion" or that the expressions are "mutually exclusive". But that is not very easy to read imho.

it might be possible to write something like:

    if 'i<n' is true then 'i==n' is always false.

we should have some such message already for some cases.

Reply:
Trying to make these warnings more readable is an old problem. See this ticket:

https://trac.cppcheck.net/ticket/4117